test/over_react/component_declaration/builder_integration_tests/new_boilerplate/function_component_test.dart

```diff
@@ -325,6 +328,91 @@ void functionComponentTestHelper(UiFactory<TestProps> factory,
+      const aRandomDataAttributeValue = 'aRandomDataAttributeValue';
+      const anAriaLabelPropValue = 'anAriaLabelPropValue';
+
+      setUp(() {
+        initialProps = (factory()
+          ..stringProp = stringProp
+          ..anotherProp = anotherProp
+          ..aRandomDataAttribute = aRandomDataAttributeValue
+          ..anAriaLabelAlias = anAriaLabelPropValue
+          ..id = idAttributeValue
+        );
+
+        secondProps = initialProps;
+      });
+
+      test('by default excludes props mixin type that it is invoked on', () {
+        TestProps unconsumedProps;
+         if (modifyProps == true) {
+          unconsumedProps = factory()..modifyProps(secondProps.addPropsToForward());
+        } else {
+          unconsumedProps = factory(secondProps.getPropsToForward());
+        }
+
+        expect(unconsumedProps.anotherProp, anotherProp);
+        expect(unconsumedProps.stringProp, isNull);
+      });
+
+      group('and props are correctly filtered', () {
+        test('for an empty set', () {
+          var unconsumedProps = _propsToForward(exclude: {}, props: initialProps, factory: factory, modifyProps: modifyProps);
+
+          expect(unconsumedProps.stringProp, stringProp);
+          expect(unconsumedProps.anotherProp, anotherProp);
+        });
+
+        test('for a single value in set', () {
+          var unconsumedProps = _propsToForward(exclude: {ASecondPropsMixin}, props: initialProps, factory: factory, modifyProps: modifyProps);
+
+          expect(unconsumedProps.stringProp, stringProp);
+          expect(unconsumedProps.anotherProp, isNull);
+        });
+
+        test('for multiple values in set', () {
+          var unconsumedProps = _propsToForward(exclude: {ASecondPropsMixin, TestPropsMixin}, props: initialProps, factory: factory, modifyProps: modifyProps);
+
+          expect(unconsumedProps.stringProp, isNull);
+          expect(unconsumedProps.anotherProp, isNull);
+        });
+
+        test('excludes dom attributes that are part of a mixin with `@Accessor` annotations ', () {
+          var unconsumedProps = _propsToForward(exclude: {DomAccessorPropsMixin}, props: initialProps, factory: factory, modifyProps: modifyProps);
+
+          expect(unconsumedProps.stringProp, stringProp);
+          expect(unconsumedProps.anotherProp, anotherProp);
+          expect(unconsumedProps.aRandomDataAttribute, isNull);
+          expect(unconsumedProps.anAriaLabelAlias, isNull);
+        });
+
+        test('for dom only ', () {
+          var unconsumedProps = _propsToForward(domOnly: true, props: initialProps, factory: factory, modifyProps: modifyProps);
+
+          expect(unconsumedProps.stringProp, isNull);
+          expect(unconsumedProps.anotherProp, isNull);
+          expect(unconsumedProps.aRandomDataAttribute, aRandomDataAttributeValue);
+          expect(unconsumedProps.anAriaLabelAlias, anAriaLabelPropValue);
+          expect(unconsumedProps.id, idAttributeValue);
```

Comment:
Can we also verify that the `id` DOM prop is forwarded or not forwarded as expected in the other test cases?

Reply:
For sure, but also I honestly don't know a case where `id` would NOT be forwarded... like unless someone manually added `id` to their props mixin.

Reply:
> I honestly don't know a case where id would NOT be forwarded.

That's true, assuming the implementation is correct 😉